### ISSUE TITLE
Block image html is same after using the plugin.

### ISSUE THREAD

**Issue by vijayhardaha** [WordPress/gutenberg#39592]
### Description

I found an issue during solving a support forum issue https://wordpress.org/support/topic/images-will-not-align-center-2/page/2/#post-15473517

So when I am not using the Gutenburg plugin and using the default WP Block editor, I create an Image Block and then I set the image and set it to align. and save it.

so in frontend I have HTML output like this
`<div class="wp-block-image"><figure class="aligncenter size-full"><img .../></figure></div>`

But then I installed the Gutenberg plugin and the go back to the old page and then update it or add any new image and update it. Then all previous block image HTML is replaced with this
`<figure class="wp-block-image aligncenter size-full"><img .../></figure>`

So this parent `div` tag is being removed and the class is merged in `figure` tag.
I am unsure if this expected behaviour of the plugin or a bug. 

### Step-by-step reproduction instructions

# Setup

- Fresh WordPress install with Twenty Twenty Two theme
- No plugin or anything else.

1. Create a New Page and then add a block image and save the page
2. Goto the Frontend and check the page Source and copy the HTML the output is like this `<div class="wp-block-image"><figure class="aligncenter size-full"><img .../></figure></div>`
3. Then Install & activate the Gutenberg plugin.
4. Then edit the old page that we just created and add another image and save it.
5. Goto frontend and check page source, this time HTML will be like this `<figure class="wp-block-image aligncenter size-full"><img .../></figure>` and not only for new image for old image as well.

### Environment info

WordPress Version: 5.9.2
Theme: Twenty Twenty Two(1.1)
Gutenberg Version: 12.8.1

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by t-hamano** [comment]
Hi. @vijayhardaha.

On #38657, the wrapper div are now removed in the theme with layout support.
("Astra" mentioned in the support forum is a classic theme that supports layouts in `theme.json`.)

Therefore, the change in HTML structure is not a bug.

However, I think that theme image block is not centered on the front-end side is a bug, so I have submitted a PR #39594.

I beleave this PR will fix the bug.
